src/installer/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs

```diff
@@ -233,6 +239,19 @@ void FindBundleHeader()
             return headerOffset != 0;
         }
 
+        private static void CodeSign(string args)
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```

Comment:
I'd throw or assert here. If this ever gets called on a non-mac, that's a bug, not a recoverable failure

Reply:
I think it's fine to assert, but I don't think we should throw. Especially since we build the entire feature as "nice to have" - it only works on mac. If I build on Linux and target mac I will have to codesign manually. So if I build on Linux and for some reason we try to codesign, it should not fail the build as the signing is not expected or it's optional.

Unfortunately we don't have a way to write to the msbuild log. Ideally we would produce at least an Info message about this so that it's easier to diagnose. I filed https://github.com/dotnet/runtime/issues/53923 to hopefully improve on this one day.

Reply:
If we eventually intend to implement codesigning on non-Mac that makes sense, but otherwise I'm actually fine with failing if we're on a Mac and the codesign tool is not found. As far as I can tell it's available on every mac back to 10.4 -- that looks like it should be plenty for all our releases of CoreCLR.